internal/auth/tokenloader.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package auth
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// CachedFileTokenLoader returns a function that loads a token from the given file path,
+// caches it for the specified interval, and reloads after the interval expires.
+func CachedFileTokenLoader(path string, interval time.Duration) func() (string, error) {
+	var (
+		mu          sync.Mutex
+		cachedToken string
+		lastRead    time.Time
+	)
+	return func() (string, error) {
+		mu.Lock()
+		defer mu.Unlock()
+		now := time.Now()
+		if cachedToken != "" && now.Sub(lastRead) < interval {
```

Comment:
if the token file happens to be empty you will be reloading it on every invocation. This is not necessary. You can use `lastRead` being zero as an indication that the load has happened and the token was cached, not the actual value of the token.